pkg/kv/kvclient/kvcoord/dist_sender_mux_rangefeed.go

```diff
@@ -179,7 +202,14 @@ func (m *rangefeedMuxer) startSingleRangeFeed(
 		startAfter:      startAfter,
 		catchupRes:      catchupRes,
 		token:           token,
+		retry:           retry.StartWithCtx(ctx, m.ds.rpcRetryOptions),
 	}
+	return m.startActiveRangeFeed(ctx, stream)
+}
+
+func (m *rangefeedMuxer) startActiveRangeFeed(
+	ctx context.Context, stream *activeMuxRangeFeed,
+) error {
 	streamID := atomic.AddInt64(&m.seqID, 1)
```

Comment:
What is the relation between `streamID` and the `stream` parameter? It feels counter-intuitive that the ID for a stream is decided after it has been passed in, and then it's passed along with the `stream` (e.g. in `startRangeFeed` method).

Can the ID change during the lifetime of the `stream`? If not, should it just be a field like `stream.ID`?

Reply:
Not only that, we want to change the stream id every time we restart range feed; it uniquely ties the server side w/ client side...

Reply:
Do I still read it correctly that a single `activeMuxRangeFeed` object can be associated only with a single `streamID`? When a rangefeed is restarted, it gets a brand new `activeMuxRangeFeed` with a new ID.

Reply:
yes.